### PR TITLE
megasync: init at 4.1.1.0

### PR DESCRIPTION
**Pull request by michojel**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by michojel** [comment]
A continuation of #53126

**Comment by michojel** [comment]
@ElXreno, @markuskowa, @matthewbauer, @NathanielHill  PTAL

File: pkgs/applications/misc/megasync/default.nix [hunk 1]
```diff
@@ -0,0 +1,143 @@
+{ lib
+, stdenv
+, autoconf
+, automake
+, bash
```
Comment (markuskowa) [inline pkgs/applications/misc/megasync/default.nix]: bash and bashInteractive are not needed.

File: pkgs/applications/misc/megasync/default.nix [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+, libzen
+, lsb-release
+, makeDesktopItem
+, pkgconfig
+, qt5
+, sqlite
+, swig
+, unzip
+, wget
+, enableFFmpeg   ? true, ffmpeg  ? ffmpeg
```
Comment (markuskowa) [inline pkgs/applications/misc/megasync/default.nix]:
```suggestion
, enableFFmpeg  ? true
, ffmpeg
```
ffmpeg does not need a default value (it is already the default package to be used).

File: pkgs/applications/misc/megasync/default.nix [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+}:
+
+assert enableFFmpeg   -> ffmpeg != null;
```
Comment (markuskowa) [inline pkgs/applications/misc/megasync/default.nix]: See previous comment. `ffmpeg` is the standard package that is used when `enableFFmpeg` is true. The assert can thus be removed.

File: pkgs/applications/misc/megasync/default.nix [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+          "--with-curl"
+          "--without-freeimage"  # unreferenced even when found
+          "--without-readline"
+          "--without-termcap"
+          "--with-sodium"
+          "--with-sqlite"
+          "--with-zlib"
+    ] ++ stdenv.lib.optionals enableFFmpeg ["--with-ffmpeg"];
+
+  # TODO: unless overriden, qmake is called instead ??
```
Comment (markuskowa) [inline pkgs/applications/misc/megasync/default.nix]: Setting `dontUseQmakeConfigure = true;` will do the job here. You can then also remove the `configureScript` line above.

File: pkgs/applications/misc/megasync/default.nix [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+  src = fetchFromGitHub {
+    owner = "meganz";
+    repo = "MEGAsync";
+    rev = "v${version}_Linux";
+    sha256 = "0lc228q3s9xp78dxjn22g6anqlsy1hi7a6yfs4q3l6gyfc3qcxl2";
+    fetchSubmodules = true;
+  };
+
+  desktopItem = makeDesktopItem {
```
Comment (markuskowa) [inline pkgs/applications/misc/megasync/default.nix]: The `desktopItem` should be outside the derivation in a let block in a pattern like this:
```
let 
  desktopItem = ... {
  };
in stdenv.mkDerivation { ...

File: pkgs/applications/misc/megasync/default.nix [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    pkgconfig
+    qt5.qmake
+    qt5.qttools
+    swig
+  ];
+  buildInputs = [
+    c-ares
+    cryptopp
+    curl
+    #freeimage    # unreferenced
```
Comment (markuskowa) [inline pkgs/applications/misc/megasync/default.nix]: That line can be removed. The comment in the configureFlags is sufficient.

**Review by markuskowa — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/megasync/default.nix (hunk 4 above)
Comment (michojel) [reply pkgs/applications/misc/megasync/default.nix]: Aha! Thanks! Added.

File: pkgs/applications/misc/megasync/default.nix (hunk 5 above)
Comment (michojel) [reply pkgs/applications/misc/megasync/default.nix]: Using a provided one now.

File: pkgs/applications/misc/megasync/default.nix (hunk 3 above)
Comment (michojel) [reply pkgs/applications/misc/megasync/default.nix]: Removed. Made `ffmpeg` no longer optional.

**Comment by michojel** [comment]
Thanks @markuskowa for the valuable feedback. Should be resolved now.

File: pkgs/applications/misc/megasync/default.nix [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+    pushd MEGASync
+      lrelease MEGASync.pro
+      DESKTOP_DESTDIR="$out" qmake PREFIX="$out" -o Makefile MEGASync.pro CONFIG+=release
+    popd
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Easy automated syncing between your computers and your MEGA Cloud Drive";
+    homepage    = https://mega.nz/;
+    license     = licenses.free;
```
Comment (markuskowa) [inline pkgs/applications/misc/megasync/default.nix]: Is `free` the right license here?
In the [license file](https://github.com/meganz/MEGAsync/blob/master/LICENCE.md) it says:
>1. Subject to the terms of this licence, we grant you a non-transferable, non-exclusive, worldwide, royalty-free licence to access and use the code solely for review purposes.
> 2. You may provide the code to anyone else and publish excerpts of it for the purposes of review and commentary [...]
>3. [...], you are not entitled to use or do anything with the code for any commercial or other purpose, other than review and commentary on it.

I am not sure if we should re-distribute the binary code built from this. Maybe we should label it as `unfree`.
CC @matthewbauer any ideas?

File: pkgs/applications/misc/megasync/default.nix [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+    unzip
+    wget
+  ];
+
+  patchPhase = ''
+    for file in $(find src/ -type f \( -iname configure -o -iname \*.sh  \) ); do
+      substituteInPlace "$file" --replace "/bin/bash" "${stdenv.shell}"
+    done
+
+    # Distro and version targets attempt to use lsb_release which is broken
```
Comment (markuskowa) [inline pkgs/applications/misc/megasync/default.nix]: The two changes below should be introduced via a patch. That will make them more maintainable in future updates.
The shell path fix above should then go into `postPatch`.

**Comment by markuskowa** [comment]
@michojel you may also consider adding `enableParallelBuilding = true` to the derivation.
@GrahamcOfBorg build megasync

File: pkgs/applications/misc/megasync/default.nix (hunk 7 above)
Comment (michojel) [reply pkgs/applications/misc/megasync/default.nix]: Ahh, good catch. "Freeware" != "free": 
https://github.com/meganz/MEGAsync/blob/master/build/templates/MEGAsync/megasync.spec#L5

Comment (michojel) [reply pkgs/applications/misc/megasync/default.nix]: Changed to `unfree`.

File: pkgs/applications/misc/megasync/default.nix (hunk 8 above)
Comment (michojel) [reply pkgs/applications/misc/megasync/default.nix]: Turned into patch files.

**Comment by michojel** [comment]
> @michojel you may also consider adding `enableParallelBuilding = true` to the derivation.

Enabled. Thanks.

**Review by markuskowa — APPROVED** [review]

**Pull request merged**